### PR TITLE
fix(mc): Selectively not copy content-src (jsx, scss) and webpack config from system-addon to mozilla-central

### PR DESCRIPTION
**Pull request by Mardak**
Address some of #2276 and #2229. r?@k88hudson 

Don't copy over content-src as the jsx files are put in the bundle and scss into the single css. Also move webpack config to the top, so it's not copied. These are to avoid linting issues in m-c. We do want to copy over the bundle, so update the top level `eslintnignore` to ignore it.

Alternative copy that picks out the desired files types (excludes jsx and scss):
`cpx "system-addon/**/{,.}*{.,_}{build,css,html,in,ini,LICENSE,md,mn,js,jsm,json}" ../mozilla-central/browser/extensions/activity-stream`

That alternate is what I started with although it seems less explicit than a list of directories/files. I suppose with the current list in the PR, I could leave the webpack config within the `system-addon` directory as it's not listed to be copied over.

### CODE REVIEW

File: mozilla-central-patches/eslintignore.diff [hunk 1]
```diff
@@ -0,0 +1,9 @@
+diff --git a/.eslintignore b/.eslintignore
+--- a/.eslintignore
++++ b/.eslintignore
+@@ -72,3 +72,4 @@ browser/extensions/pocket/content/panels/js/vendor/**
+ browser/locales/**
+-# vendor library files in activity-stream
++# generated or library files in activity-stream
++browser/extensions/activity-stream/data/content/activity-stream.bundle.js
```
Comment (Mardak) [inline mozilla-central-patches/eslintignore.diff]: Are we expecting other bundles? Or js files in that data/content directory? Maybe just ignore that whole directory? The explicit `bundles` subdirectory would make it clearer why it's being ignored though.

File: webpack.system-addon.config.js [hunk 2]
```diff
@@ -18,9 +19,9 @@ module.exports = {
   },
   // This resolve config allows us to import with paths relative to the system-addon/ directory, e.g. "lib/ActivityStream.jsm"
   resolve: {
+    extensions: [".jsx"],
```
Comment (Mardak) [inline webpack.system-addon.config.js]: This would be to include the `lib` files, yeah? I'll add that in.

File: webpack.system-addon.config.js [hunk 3]
```diff
@@ -18,9 +19,9 @@ module.exports = {
...
     modules: [
-      __dirname,
-      "node_modules"
```
Comment (Mardak) [inline webpack.system-addon.config.js]: Okay, restoring `node_modules`

**Comment by as-pine-proxy** [comment]
[Treeherder: 838c5174dfab6ca4fe1a0e70f43ba7aa929746c6](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=838c5174dfab6ca4fe1a0e70f43ba7aa929746c6)

**Comment by Mardak** [comment]
greeeen! https://treeherder.mozilla.org/#/jobs?repo=pine&revision=838c5174dfab6ca4fe1a0e70f43ba7aa929746c6&selectedJob=89723792&filter-searchStr=eslint

**Comment by k88hudson** [comment]
\o/ 🍏 📗 💚 🍏 📗 💚 🍏 📗 💚 🍏 📗 💚

**Pull request merged**